### PR TITLE
Remove all non-functional/non-informational uses of mscorlib.

### LINKED ISSUE #7358: Remove mscorlib references from CoreCLR
After https://github.com/dotnet/coreclr/pull/9278 goes in, the term "mscorlib" is not relevant to CoreCLR.

We can do some reasonable cleanup of this. I don't think it is worthwhile to remove it from every define, comment, etc but certainly from file and folder names and do a futher grep to see if there is other worthwhile instances.

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Fixes #7358

/cc @MichalStrehovsky @danmoseley

### CODE REVIEW

File: src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -42,7 +42,7 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <!-- This prevents the default MsBuild targets from referencing System.Core.dll -->
     <AddAdditionalExplicitAssemblyReferences>false</AddAdditionalExplicitAssemblyReferences>
-    <!-- These prevent the default MsBuild targets from referencing System.dll and mscorlib.dll -->
+    <!-- These prevent the default MsBuild targets from referencing System.dll and System.Private.CoreLib.dll -->
```
Comment (jkotas) [inline src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:45]: Is `ExcludeMscorlibFacade` actually used anywhere? I am not able to find it in any target files.

File: src/coreclr/tools/aot/ILCompiler.Diagnostics/ISymNGenWriter.cs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#pragma warning disable 436 // SuppressUnmanagedCodeSecurityAttribute defined in source and mscorlib
+#pragma warning disable 436 // SuppressUnmanagedCodeSecurityAttribute defined in source and SPCL
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.Diagnostics/ISymNGenWriter.cs:4]: Is this pragma disable still relevant? Can we just delete it?

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Augments/ReflectionAugments.cs [hunk 3]
```diff
@@ -12,7 +12,7 @@
 //
 //  Implemented by:
 //    System.Reflection.dll on RH (may use ILMerging instead)
-//    mscorlib.dll on desktop
+//    System.Private.CoreLib.dll on CoreCLR
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Reflection/Augments/ReflectionAugments.cs:15]:
```suggestion
```
This comment does not make sense

File: src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj (hunk 1 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:45]: I looked too. Based on the comments, I was assuming it was in the official MsBuild targets. Let me do some more digging.

Comment (AaronRobinsonMSFT) [reply src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj:45]: I've looked under the dotnet organization as well as under microsoft/msbuild - no mentioned anywhere. I'll remove both properties.

File: src/coreclr/tools/aot/ILCompiler.Diagnostics/ISymNGenWriter.cs (hunk 2 above)
Comment (AaronRobinsonMSFT) [reply src/coreclr/tools/aot/ILCompiler.Diagnostics/ISymNGenWriter.cs:4]: Ugh. Yes, this should just be removed.

**Review by jkotas — APPROVED** [review]

**Pull request merged**